### PR TITLE
fix(stream): remove unused last_watermark in state_table

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

1. `last_watermark` is always `None`, it's not maintained correctly.
2. We don't need `last_watermark` now.

## Checklist For Contributors

- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/stream/src/common/table/state_table.rs [hunk 1]
```diff
@@ -832,19 +836,13 @@ impl<S: StateStore> StateTable<S> {
             }
         }
         if let Some(range_end_suffix) = range_end_suffix {
-            let range_begin_suffix = if let Some(ref last_watermark) = self.last_watermark {
```
Comment (soundOfDestiny) [inline src/stream/src/common/table/state_table.rs:835]: @Little-Wallace told me to write this part of code. If you want to remove it, I recommend to discuss with him.

**Review by soundOfDestiny — APPROVED** [review]
rest LGTM

File: src/stream/src/common/table/state_table.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/stream/src/common/table/state_table.rs:835]: We only need to remove all data before some watermark such as (-∞, t]

Comment (soundOfDestiny) [reply src/stream/src/common/table/state_table.rs:835]: > We only need to remove all data before some watermark such as (-∞, t]

You said that you preferred a lower bound though your algorithm can work without lower bound.

**Pull request merged**